kroxylicious-api/src/main/java/io/kroxylicious/proxy/filter/FilterFactoryContext.java

```diff
@@ -27,6 +30,8 @@ public interface FilterFactoryContext {
      */
     ScheduledExecutorService eventLoop();
 
+    SSLContext clientSslContext(Tls tls);
```

Comment:
If we tie the API to JDK SslContext, we prevent the user taking advantage of OpenSSL.  Maybe that's good enough for now.